packages/mdc-tab-bar/foundation.js

```diff
@@ -0,0 +1,365 @@
+/**
+  * @license
+  * Copyright 2018 Google Inc. All Rights Reserved.
+  *
+  * Licensed under the Apache License, Version 2.0 (the "License")
+  * you may not use this file except in compliance with the License.
+  * You may obtain a copy of the License at
+  *
+  *      http://www.apache.org/licenses/LICENSE-2.0
+  *
+  * Unless required by applicable law or agreed to in writing, software
+  * distributed under the License is distributed on an "AS IS" BASIS,
+  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  * See the License for the specific language governing permissions and
+  * limitations under the License.
+*/
+
+import MDCFoundation from '@material/base/foundation';
+
+import {strings, numbers} from './constants';
+import MDCTabBarAdapter from './adapter';
+
+/* eslint-disable no-unused-vars */
+import MDCTabFoundation from '@material/tab/foundation';
+import {MDCTabDimensions} from '@material/tab/adapter';
+/* eslint-enable no-unused-vars */
+
+/**
+ * @extends {MDCFoundation<!MDCTabBarAdapter>}
+ * @final
+ */
+class MDCTabBarFoundation extends MDCFoundation {
+  /** @return enum {string} */
+  static get strings() {
+    return strings;
+  }
+
+  /** @return enum {number} */
+  static get numbers() {
+    return numbers;
+  }
+
+  /**
+   * @see MDCTabBarAdapter for typing information
+   * @return {!MDCTabBarAdapter}
+   */
+  static get defaultAdapter() {
+    return /** @type {!MDCTabBarAdapter} */ ({
+      scrollTo: () => {},
+      incrementScroll: () => {},
+      getScrollPosition: () => {},
+      getScrollContentWidth: () => {},
+      getOffsetWidth: () => {},
+      isRTL: () => {},
+      activateTabAtIndex: () => {},
+      deactivateTabAtIndex: () => {},
+      getTabIndicatorClientRectAtIndex: () => {},
+      getTabDimensionsAtIndex: () => {},
+      getActiveTabIndex: () => {},
+      getIndexOfTab: () => {},
+      getTabListLength: () => {},
+    });
+  }
+
+  /**
+   * @param {!MDCTabBarAdapter} adapter
+   * */
+  constructor(adapter) {
+    super(Object.assign(MDCTabBarFoundation.defaultAdapter, adapter));
+  }
+
+  init() {
+    const activeIndex = this.adapter_.getActiveTabIndex();
+    this.scrollIntoView(activeIndex);
+  }
+
+  /**
+   * Activates the tab at the given index
+   * @param {number} index
+   */
+  activateTab(index) {
+    const previousActiveIndex = this.adapter_.getActiveTabIndex();
+    if (!this.indexIsInRange_(index) || index === previousActiveIndex) {
+      return;
+    }
+
+    this.adapter_.deactivateTabAtIndex(previousActiveIndex);
+    this.adapter_.activateTabAtIndex(index, this.adapter_.getTabIndicatorClientRectAtIndex(previousActiveIndex));
+    this.scrollIntoView(index);
+  }
+
+  /**
+   * Handles the keydown event
+   * @param {!Event} evt
+   */
+  handleKeyDown(evt) {
+    const keyboardNavigation = [
+      'ArrowLeft',
```

Comment:
IE and Edge report `Left` and `Right` instead of `ArrowLeft` and `ArrowRight`, so we need to handle those here and in `activateTabFromKeydown_`.